src/fetcher.py

```diff
@@ -56,6 +56,8 @@ def dump_kaikki_dict(
     @staticmethod
     @functools.lru_cache
     def _get_word_json(dict_dir: Path, word: str) -> dict:
+        # TODO: handle words with multiple word senses
```

Comment:
I don't understand this comment. Should not multiple word senses be already handled by get_senses? Or maybe you have something else in mind?

Reply:
Sorry, it is for something else - I'm thinking of a way to handle words with multiple definitions (it is called "word senses" in kaikki.org), e.g., [Kohl](https://kaikki.org/dictionary/German/meaning/K/Ko/Kohl.html#Kohl-de-noun-UyDRBzbl) has 3: noun (cabbage), noun (non-sense), proper name

Reply:
I see. I was not aware of this. The `dump_kaikki_dict` function can be modified to append the word json to the word file instead of overwriting its previous contents. Then maybe `_get_word_json` could return a list of dicts for all senses. If you're interested in working on this let me know, otherwise I'll add it to my todo list.

Reply:
With the following change in `dump_kaikki_dict`, it can save different definitions into indifidual JSON files:

```python

            for i, line in enumerate(file):
                entry = json.loads(line)
                word = entry["word"]

                # if the filename exists, rename to {word}_00.json (or _01, _02 etc.)
                outfile = f"{word}.json"
                nameidx = 0
                while (Path(outdir / outfile).is_file()):
                    outfile = f"{word}_{nameidx:02}.json"
                    nameidx += 1

                try:
                    with open(
                        outdir / outfile,
                        mode="w",
                        encoding="utf-8",
                    ) as outfile:
```

The problem is how to let the user to select the definition they wants. I'm thinking of a text box and arrow buttons (like below), but it may take some time because I've never use Qt before.

![Screenshot_2022-12-22_03-17-15](https://user-images.githubusercontent.com/23568324/209041444-48d2c161-e6fd-4047-9ac7-d32e434fc4e7.png)

Reply:
> The problem is how to let the user to select the definition they wants.

What about just dumping all definitions to the notes but improving the formatting so that it's clear to which sense each definition/example/etc belongs?

> I'm thinking of a text box and arrow buttons

I guess this will only work or be easy to implement when a single note is processed, not when multiple notes are processed in bulk from the browser.

Reply:
Another option is to maybe introduce a mode where the add-on generates new notes for each sense and fill them instead of working only on existing notes.

Reply:
> not when multiple notes are processed in bulk from the browser.

I don't know about that, is that a feature of Anki?

Reply:
> I don't know about that, is that a feature of Anki?

No, it's a feature of this add-on. It can process multiple notes at once if you select some notes in the browser and open the add-on's window from *Edit > Bulk-define from Wiktionary*.